spec/models/tour_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'rails_helper'
+
+RSpec.describe Tour, type: :model do
+
+  describe "validatins" do
+    it { should validate_presence_of(:tour_name) }
+    it { should validate_presence_of(:image) }
+  end
+
+  describe "associations" do
+    it { should belong_to(:tour_category)}
+    it { should have_many(:bookings).through(:tour_details).source(:bookings) }
+    it { should have_many(:tour_followings).dependent(:destroy) }
+    it { should have_many(:followed_users).through(:tour_followings).source(:user) }
+    it { should have_many(:reviews).through(:tour_details).source(:reviews) }
+  end
+
+  describe "scopes" do
+    it "orders tour by creation date in descending order" do
+      new_tour1 = create(:tour)
+      new_tour2 = create(:tour)
```

Comment:
Mấy tk phía dưới anh thấy cũng tạo giống nhưu này thì nên sử dụng let!. Với lại tìm hiểu và phân biệt cho anh let vs let! luôn nhé

Reply:
let : let khai báo biến trong khối và có giá trị trong từng example ( mỗi example ("it" block) sẽ khởi tạo lại )
let! : let! tương tự như let dùng để khai báo biến nhưng sẽ chỉ khởi tạo lại lần đầu (= việc chạy 1 before ngầm)